### DISCUSSION TITLE
Is there a way to partially validate form?

### DISCUSSION THREAD

**Discussion by Steffi3rd** [react-hook-form/react-hook-form#3676] [Q&A] [answered]
Hello, 

There is `formState.isValid` which valids the entire form. I already use it at the end on my form:
```
<Button disabled={!formState.isValid}>Submit</Button>
```

I would like to have something similar but with partial validation with defined fields.
I would use it in the middle of the form. 

For example : I have 6 required fields. (firstName, middleName, lastName, email, phone, website)
I would like to disable the first 3 fields (`firstName`, `middleName`, `lastName`) are not valid yet. 

```
<form>

   // firstName 
   // middleName
   // lastName

   <Button disabled={ ... ??? ...} type="button" onClick={doSomething}>Continue</Button>

   // email 
   // phone
   // website

   <Button disabled={!formState.isValid} type="submit">Submit</Button>

</form>
```

Is there a way to do that please?

**Comment by kanu-shubham** [comment, marked as answer]
You could use getValues and check if those three fields have value or not and set the button as disabled.

**Reply by Steffi3rd** [reply]
Thanks! Yes, I could but in my case _have value_ is not enough because some fields have diffrerents rules to be valid. That’s why I wanted similar to `formState.isValid`

**Reply by bluebill1049** [reply]
@kanu-shubham is correct, you will need to leverage `validate` with `getValues` to dynamically change validation based on xxx

**Reply by Steffi3rd** [reply]
Do you have a little snippet on code both could work together please?

**Reply by bluebill1049** [reply]
here is one: https://codesandbox.io/s/react-hook-form-conditional-fields-forked-n0jig?file=/src/index.js:0-1831

**Reply by Steffi3rd** [reply]
Thank you guys!
